packages/components/app/styles/components/dialog-primitive.scss

```diff
@@ -0,0 +1,119 @@
+/**
+ * Copyright (c) HashiCorp, Inc.
+ * SPDX-License-Identifier: MPL-2.0
+ */
+
+//
+// DIALOG PRIMITIVE
+//
+
+
+// MAIN CONTAINER (<dialog>)
+
+// :where reduces the specificity of the selector
+// & can be used to set “loose” defaults which can be easily overridden
+dialog:where(.hds-dialog-primitive) {
+  all: unset; // removes default dialog & inherited styles. Must be the first style to be applied
```

Comment:
Using `all: unset` causes inherited styles such as border-box styling to be wiped out and also wipes out some default `dialog` styles the `Modal` & `Flyout` were relying upon so may still want to consider changing to use a `div` to control the layout vs. using `dialog`.

Let me know what you think @didoo

Reply:
conversation continued here: https://github.com/hashicorp/design-system/pull/1918/files#r1482873567